File: samples/java-app/src/main/java/com/appambit/javaapp/models/Post.java
package com.appambit.javaapp.models;

import com.appambit.sdk.utils.JsonKey;

public class Post {
    public String id;
    public String title;
    public String body;
    public String category;
    public String author;

    @JsonKey("featured_image")
    public String featuredImage;

    public int likes;

    public double rating;

    @JsonKey("reading_time")
    public int readingTime;

    @JsonKey("published_at")
    public String publishedAt;
}

File: samples/kotlin-app/src/main/java/com/appambit/kotlinapp/models/Post.kt
package com.appambit.kotlinapp.models

import com.appambit.sdk.utils.JsonKey

class Post {
    @JvmField var id: String? = null
    @JvmField var title: String? = null
    @JvmField var body: String? = null
    @JvmField var category: String? = null
    @JvmField var author: String? = null

    @JvmField
    @JsonKey("featured_image")
    var featuredImage: String? = null

    @JvmField var likes: Int = 0
    @JvmField var rating: Double = 0.0

    @JvmField
    @JsonKey("reading_time")
    var readingTime: Int = 0

    @JvmField
    @JsonKey("published_at")
    var publishedAt: String? = null
}
